packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -190,7 +190,7 @@ where
             .into_report()
             .map(|row| row.get(0))
             .map_err(|report| match report.current_context().code() {
-                Some(&SqlState::UNIQUE_VIOLATION) => report
+                Some(&SqlState::EXCLUSION_VIOLATION | &SqlState::UNIQUE_VIOLATION) => report
```

Comment:
When does `EXCLUSION_VIOLATION` happen for us? I see in the docs it can happen due to serialization errors, so is it to do with the function call handing off to the trigger?

Reply:
If you have a type with `id=1` with the time of `[10, ∞)` and then try to insert a new type with `id=1` the time intervals will overlap, which is not allowed. So this check ensures, that the type does not exist already in the database.

Reply:
So it happens when the GIST index is violated?

Reply:
Yup 🙂